subsys/bluetooth/host/addr.c

```diff
@@ -53,28 +53,30 @@ int bt_addr_le_create_static(bt_addr_le_t *addr)
 
 int bt_addr_from_str(const char *str, bt_addr_t *addr)
 {
-	int i, j;
-	uint8_t tmp;
+	const size_t strlen_addr = strlen("XX:XX:XX:XX:XX:XX");
 
-	if (strlen(str) != 17U) {
+	if (strlen(str) != strlen_addr) {
 		return -EINVAL;
 	}
 
-	for (i = 5, j = 1; *str != '\0'; str++, j++) {
-		if (!(j % 3) && (*str != ':')) {
-			return -EINVAL;
-		} else if (*str == ':') {
-			i--;
-			continue;
-		}
+	for (size_t i = 0; i < strlen_addr; i++) {
+		int octet = i / 3; /* Count whole octets 'ab:' in 'ab:ab:ab:ab:ab:ab'. */
+		int digit = i % 3; /* Cycle over indexes in 'ab:'. */
```

Comment:
```suggestion
		const size_t octet = i / 3; /* Count whole octets 'ab:' in 'ab:ab:ab:ab:ab:ab'. */
		const size_t digit = i % 3; /* Cycle over indexes in 'ab:'. */
```

I would make these `const size_t` as they will not change and the source value for the calculations is also a `size_t` (`i`).